### PR TITLE
implemented getStatisctics() method

### PR DESCRIPTION
**Pull request by CodeWebWeaver**
implemented 3 private additional methods to:
 - read file
 - create report
 - write to file

### CODE REVIEW

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 1]
```diff
@@ -1,7 +1,75 @@
...
+            if (!bufferedReader.ready()) {
+                return gatheredData.toString();
+            }
+            String value = bufferedReader.readLine();
+            while (value != null) {
+                gatheredData.append(value).append(System.lineSeparator());
+                value = bufferedReader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can`t read file", e);
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/WorkWithFile.java:35]: add filename

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 2]
```diff
@@ -1,7 +1,75 @@
...
+        }
+        return gatheredData.toString();
+    }
+
+    private StringBuilder buildReport(String unformattedData) {
+        System.out.println(unformattedData);
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/WorkWithFile.java:41]:
```suggestion
```

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 3]
```diff
@@ -1,7 +1,75 @@
...
+        int supply = 0;
+        int buy = 0;
+        String[] gatheredData = unformattedData.split(System.lineSeparator());
+        int commaIndex;
+        for (String dataLine : gatheredData) {
+            commaIndex = dataLine.indexOf(COMA_SEPARATOR);
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/WorkWithFile.java:47]: let's better split line by comma

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 4]
```diff
@@ -1,7 +1,75 @@
...
+                .append(supply - buy).append(System.lineSeparator());
+        return report;
+    }
 
+    private void writeToFile(String toFileName, String data) {
+        File file = new File(toFileName);
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(file))) {
+            bufferedWriter.write(data);
+        } catch (IOException e) {
+            throw new RuntimeException("Can`t write to file", e);
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/WorkWithFile.java:72]: add filename

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 5]
```diff
@@ -1,7 +1,72 @@
...
+
+    private StringBuilder buildReport(String unformattedData) {
+        int supply = 0;
+        int buy = 0;
+        String[] gatheredData = unformattedData.split(System.lineSeparator());
+        String[] splitLine;
+        for (String dataLine : gatheredData) {
+            splitLine = dataLine.split(COMA_SEPARATOR);
+            if (splitLine[0].equals(BUY_OPERATION)) {
+                buy += Integer.parseInt(splitLine[1]);
```
Comment (nacenik) [inline src/main/java/core/basesyntax/WorkWithFile.java:48]: create constants for array indexes

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 6]
```diff
@@ -1,7 +1,72 @@
...
+            }
+            if (splitLine[0].equals(SUPPLY_OPERATION)) {
+                supply += Integer.parseInt(splitLine[1]);
+            }
```
Comment (nacenik) [inline src/main/java/core/basesyntax/WorkWithFile.java:52]: do you really need to have 2 separate if block? What about if-else?

**Review by nacenik — APPROVED** [review]
Great job! You can improve your solution if you want 😉

**Pull request closed without merging**